keras/callbacks.py

```diff
@@ -2733,6 +2774,13 @@ def on_train_batch_end(self, batch, logs=None):
                 1.0 / batch_run_time,
                 step=self._train_step,
             )
+
+        # `logs` is a `RemoteValue` when using asynchronous strategies, for now
+        # we just disable `update_freq` entirely in those cases.
+        if isinstance(logs, dict):
```

Comment:
Do we need to only do this when it's at the right batch number if `update_freq` is an integer?

Reply:
`_push_writer()` already makes it so that `update_freq` applies to all batch logs (including the `write_steps_per_second` block above), see here: https://github.com/keras-team/keras/blob/86e9ed51317872e1dfc75b03c1e257638bd7b083/keras/callbacks.py#L2633-L2646